### DISCUSSION TITLE
help with RF configuration for cabled setup

### DISCUSSION THREAD

**Discussion by rob-the-dude** [srsran/srsRAN_4G#1349] [Q&A]
Caveat: I'm mostly a software guy, with somewhat limited understanding of RF.

I'm trying to set up a test LTE network.  The primary goal is to use the test SDR network to allow me to test a cellular chipset (Quectel EC200U-CN) in the US -- this chipset doesn't support the bands that US carriers provide, but I'm developing an IoT system for a different market.  (Also, I'm not particularly interested in the RF-characteristics at the moment -- I'm actually trying to get higher-level functions -- IP connectivity -- working.)

In particular, this chipset supports LTE Band 3, which is available in the regions we intend to support.

As an SDR, I have a Lime SDR Mini 2.x (I think 2.4 - just bought it a couple weeks ago).  I also have access to a HackRF-One if that helps anything.

I have a Linux PC running mainline code for LimeSuite, SoapySDR, and srsRAN_4G (which I compiled from source).

I burned a SIM card with various parameters and put them into the user_db.csv.  That SIM card is in the EC200U module which is connected via a UART to my development system, and I'm running AT commands in a terminal program.

Because I'm in the US, I don't want to run the setup in the open (pretty sure it's illegal to do that).  So I have everything connected via a XRDS-RF 698-2700 MHz splitter/combiner that I got from Amazon (https://www.amazon.com/dp/B0B3MKTWFL).  (I also have access to a Mini-Circuits ZAPD-23-S+ if that helps.)  The Lime board has its TX and RX antennas connected via 4' cables into the two "split" ports on the splitter/combiner.  The EC200 module's antenna is cabled with a 30dB attenuator to the "input" side of the splitter/combiner.

I'm running "srsepc" and "srsenb" and they LOOK like they're working correctly.  Here are the CHANGED bits from configuration files (everything else is from the default):

```
	enb.conf (changed parameters only):

		mcc = 999
		mnc = 999
		tx_gain = 66
		rx_gain = 47

		device_name = soapy
		device_args = driver=lime,rxant=LNAH,txant=BAND2

	rr.conf (changed parameters only)

		dl_earfcn = 1200;

	epc.conf (changed parameters only)

		mcc = 999
		mnc = 999
		apn = mynetwork

```	

Here's the output when I run them:

```
	> srsepc epc.conf

	Built in Release mode using commit ec29b0c1f on branch master.

	---  Software Radio Systems EPC  ---

	Reading configuration file epc.conf...
	HSS Initialized.
	MME S11 Initialized
	MME GTP-C Initialized
	MME Initialized. MCC: 0xf999, MNC: 0xf999
	SPGW GTP-U Initialized.
	SPGW S11 Initialized.
	SP-GW Initialized.
	Received S1 Setup Request.
	S1 Setup Request - eNB Name: srsenb01, eNB id: 0x19b
	S1 Setup Request - MCC:999, MNC:999
	S1 Setup Request - TAC 7, B-PLMN 0x999999
	S1 Setup Request - Paging DRX v128
	Sending S1 Setup Response

```

```
	> srsenb enb.conf

	Active RF plugins: libsrsran_rf_soapy.so
	Inactive RF plugins:
	---  Software Radio Systems LTE eNodeB  ---

	Reading configuration file enb.conf...
	WARNING: cpu0 scaling governor is not set to performance mode. Realtime processing could be compromised. Consider setting it to performance mode before running the application.

	Built in Release mode using commit ec29b0c1f on branch master.

	Opening 1 channels in RF device=soapy with args=driver=lime,rxant=LNAH,txant=BAND2
	Supported RF device list: soapy file
	Soapy has found device #0: addr=24607:1027, driver=lime, label=LimeSDR Mini [USB 3.0] 1DA13B18FBAC83, media=USB 3.0, module=FT601, name=LimeSDR Mini, serial=1DA13B18FBAC83,
	Selecting Soapy device: 0
	[INFO] Make connection: 'LimeSDR Mini [USB 3.0] 1DA13B18FBAC83'
	[INFO] Reference clock 40.00 MHz
	[INFO] Device name: LimeSDR-Mini_v2
	[INFO] Reference: 40 MHz
	[INFO] LMS7002M register cache: Disabled
	Setting up Rx stream with 1 channel(s)
	Setting up Tx stream with 1 channel(s)
	[INFO] RX LPF configured
	[INFO] Filter calibrated. Filter order-4th, filter bandwidth set to 5 MHz.Real pole 1st order filter set to 2.5 MHz. Preemphasis filter not active
	[INFO] TX LPF configured
	Available device sensors:
	 - clock_locked
	 - lms7_temp
	Available sensors for Rx channel 0:
	 - lo_locked
	Setting Rx channel 0 antenna to LNAH
	Setting Tx channel 0 antenna to BAND2
	State of gain elements for Rx channel 0 (AGC not supported):
	 - TIA: 9.00 dB
	 - LNA: 30.00 dB
	 - PGA: -7.00 dB
	State of gain elements for Tx channel 0 (AGC not supported):
	 - PAD: 0.00 dB
	 - IAMP: 0.00 dB
	Rx antenna set to LNAH
	Tx antenna set to BAND2

	==== eNodeB started ===
	Type <t> to view trace
	[INFO] RX LPF configured
	[INFO] Filter calibrated. Filter order-4th, filter bandwidth set to 11.52 MHz.Real pole 1st order filter set to 2.5 MHz. Preemphasis filter not active
	[INFO] TX LPF configured
	Setting frequency: DL=1805.0 Mhz, UL=1710.0 MHz for cc_idx=0 nof_prb=50
	[INFO] Tx calibration finished
	[INFO] Rx calibration finished


```
But when I send my AT commands to the modem, it never seems to see a network, much less attempt to register with it ("AT+COPS=?" returns an error indicating no networks available).

I've also tried enabling packet captures via the enb.conf file, but nothing is written to these files (they're all zero-length) which makes me suspect no packets are actually getting there.

And that has me suspicious of the RF hardware and software configuration.

1) Does the 30 dB attenuator seem right?  Should it be more or less?

2) Do the TX and RX gain values I've put in (taken from an example on the web) seem correct?  I have no idea how to calculate either of these things.

3) Anything else that folks would recommend I try?

Thanks a ton for any help.

**Comment by epena429** [comment]
Where you able to get anywhere with this I'm running a similiar setup and I'm receiving the same issues.
